cartographer/mapping/pose_extrapolator.h

```diff
@@ -49,6 +49,16 @@ class PoseExtrapolator {
   void AddOdometryData(const sensor::OdometryData& odometry_data);
   transform::Rigid3d ExtrapolatePose(common::Time time);
 
+  // Must be called after the first IMU data and no pose were added. Estimates
+  // an initial gravity aligned orientation from the linear acceleration of the
+  // first IMU data.
+  void AddInitialGravityAlignedPose();
```

Comment:
Shouldn't this be called in 2D as well? We also use it for gravity alignment there I think?

Also, at that point, doesn't it make sense to just require that, if IMU data is added, it must be added first so that this can be done as the first action? Maybe taking one reading in the constructor or something?